resolwe_bio/processes/alignment/star.py

```diff
@@ -0,0 +1,704 @@
+            attributes = inputs.output_options.sam_attributes.split(" ")
+            star_params.extend(["--outSAMattributes", attributes])
+
+        if inputs.output_options.rg_line:
+            star_params.extend(["--outSAMattrRGline", inputs.output_options.rg_line])
+
+        return_code, _, _ = Cmd["STAR"][star_params] & TEE(retcode=None)
+        log_file = Path("Log.out")
+        if log_file.is_file():
+            outputs.log = str(log_file)
+        if return_code:
+            self.error("Reads alignment failed.")
+
+        star_unmapped_r1 = Path("Unmapped.out.mate1")
+        if star_unmapped_r1.is_file():
+            unmaped_out_1 = f"{mate1_name}_unmapped.out.mate1.fastq"
+            star_unmapped_r1.rename(unmaped_out_1)
+
+            return_code, _, _ = Cmd["pigz"][unmaped_out_1] & TEE(retcode=None)
+            if return_code:
+                self.error("Compression of unmapped mate 1 reads failed.")
+
+            outputs.unmapped_1 = f"{unmaped_out_1}.gz"
+
+        star_unmapped_r2 = Path("Unmapped.out.mate1")
+        if (
+            inputs.reads.type.startswith("data:reads:fastq:paired:")
+            and star_unmapped_r2.is_file()
+        ):
+            unmaped_out_2 = f"{mate2_name}_unmapped.out.mate2.fastq"
+            star_unmapped_r2.rename(unmaped_out_2)
+
+            return_code, _, _ = Cmd["pigz"][unmaped_out_2] & TEE(retcode=None)
+            if return_code:
+                self.error("Compression of unmapped mate 2 reads failed.")
+            outputs.unmapped_2 = f"{unmaped_out_2}.gz"
+
+        self.progress(0.8)
+
+        aligned_reads = (
+            "Aligned.sortedByCoord.out.bam" if inputs.star_sort else "Aligned.out.bam"
+        )
+        out_bam = f"{mate1_name}.bam"
+        out_bai = f"{out_bam}.bai"
+
+        sort_params = [
+            aligned_reads,
+            "-o",
+            out_bam,
+            "-@",
+            self.requirements.resources.cores,
+        ]
+        return_code, _, _ = Cmd["samtools"]["sort"][sort_params] & TEE(retcode=None)
+        if return_code:
+            self.error("Samtools sort command failed.")
+
+        outputs.bam = out_bam
+
+        return_code, _, _ = Cmd["samtools"]["index"][out_bam, out_bai] & TEE(
+            retcode=None
+        )
+        if return_code:
+            self.error("Samtools index command failed.")
+
+        outputs.bai = out_bai
+
+        self.progress(0.9)
+
+        if inputs.detect_chimeric.chimeric:
+            out_chimeric = f"{mate1_name}_chimeric.out.sam"
+            Path("Chimeric.out.sam").rename(out_chimeric)
+            outputs.chimeric = out_chimeric
+
+        if inputs.t_coordinates.quant_mode:
+            out_transcriptome = f"{mate1_name}_aligned.toTranscriptome.out.bam"
+            Path("Aligned.toTranscriptome.out.bam").rename(out_transcriptome)
+            outputs.alignment_transcriptome = out_transcriptome
+
+        out_stats = f"{mate1_name}_stats.txt"
```

Comment:
Are we absolutely sure these files will be created and we're not testing for their existence as for the rest?

Reply:
This should be created if the STAR exited with non-zero exit status. We are testing for other files because they may not be created if there are no such reads detected.